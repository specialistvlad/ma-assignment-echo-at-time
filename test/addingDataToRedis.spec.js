const FakeRedis = require('./utils/FakeRedis');
const EchoQueue = require('./utils/QueueWithMockedVariables');

describe('addingDataToStorage', () => {
  it('check positive flow of quering to redis', async () => {
    const dataSet = [
      [4000, 'Message 3'],
      [6000, 'Message 5'],
      [1000, 'Message 1'],
      [5001, 'Message 4'],
      [5002, 'Message 4_2'],
      [3000, 'Message 2']
    ];

    const storage = new FakeRedis();
    const echoQueue = new EchoQueue({ storage });
    await Promise.all(dataSet.map(item => echoQueue.add(...item)));
    expect(storage.history).toMatchSnapshot();
  });
});
