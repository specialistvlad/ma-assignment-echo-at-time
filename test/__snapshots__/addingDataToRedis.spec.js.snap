// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`addingDataToStorage check positive flow of quering to redis 1`] = `
Array [
  Object {
    "args": Array [
      4000,
      "not random string generator:Message 3",
    ],
    "command": "zadd",
  },
  Object {
    "args": Array [
      6000,
      "not random string generator:Message 5",
    ],
    "command": "zadd",
  },
  Object {
    "args": Array [
      1000,
      "not random string generator:Message 1",
    ],
    "command": "zadd",
  },
  Object {
    "args": Array [
      5001,
      "not random string generator:Message 4",
    ],
    "command": "zadd",
  },
  Object {
    "args": Array [
      5002,
      "not random string generator:Message 4_2",
    ],
    "command": "zadd",
  },
  Object {
    "args": Array [
      3000,
      "not random string generator:Message 2",
    ],
    "command": "zadd",
  },
]
`;
